### ISSUE TITLE
Bind BASS effects

### ISSUE THREAD

**Issue by wyozi** [Facepunch/garrysmod-requests#63]
According to http://www.un4seen.com/bass.html BASS sounds can have effects/custom DSP applied to them. It would be nice to have access to these.

**Comment by Grocel** [comment]
Yeah it would be nice to have them, but I don't think so that these will be in the game. According to the documentation they seem to be Windows/DirectX exclusive or at least bitchy on other platforms.

**Comment by wyozi** [comment]
http://www.un4seen.com/doc/#bass/effects.html

>  Away from Windows, the DX8 effects are emulated by BASS and the "With FX flag" system is unavailable. Floating-point is supported (8.24 fixed-point on Android and Windows CE), and the PARAMEQ effect also supports more than stereo.

**Comment by Grocel** [comment]
Then 1+ support from me.

**Comment by wyozi** [comment]
Friendly bump. Having reverb is almost vital for getting the true immersion within some areas.

Here's a Lua API suggestion:
```lua
IGmodAudioChannel:SetEffect(BASSFX_REVERB, {
  fInGain= 0,
  fReverbMix = 0,
  fReverbTime = 0,
  fHighFreqRTRatio = 0
})
IGmodAudioChannel:ResetEffect()
```

**Comment by wyozi** [comment]
Celebrating almost five years of being open by submitting a reference to how LÖVE2D does filters/effects:

https://love2d.org/wiki/Source:setFilter
https://love2d.org/wiki/Source:setEffect

**Comment by KrzOkun** [comment]
Friendly reminder

**Comment by Grocel** [comment]
> If this is added, it would also be useful to possibly have a bind for this added too. https://www.un4seen.com/doc/#bass/BASS_ATTRIB_TAIL.html

I think that would require Bass to be updated to latest version.
https://github.com/Facepunch/garrysmod-requests/issues/1885

**Comment by RaphaelIT7** [comment]
> > If this is added, it would also be useful to possibly have a bind for this added too. https://www.un4seen.com/doc/#bass/BASS_ATTRIB_TAIL.html
> 
> I think that would require Bass to be updated to latest version. [#1885](https://github.com/Facepunch/garrysmod-requests/issues/1885)

It should exist already since `1.2` but `2.4.16` did include a fix related to it.

**Comment by N3xsX** [comment]
Adding my support as well. I'm currently working on a music-based addon and being able to apply even basic parametric EQ or reverb in-game would vastly expand creative possibilities
